---
layout: page
title: Genomic databases and variant annotation
summary: "Genomic databases and variant annotation"
sidebar_link: true
---

# Variant annotation

-	Once called, variants (SNPs and InDels) need to be annotated
-	We want to know for example if a variant is located in a gene, if itâ€™s in the coding portion of that gene, if it causes an aminoacid substitution, if that substitution is deleterious for the encoded protein function.
-	**Variant annotation** is the process of attaching biological information from multiple available source (**public databases**) to variants
---
### Gene model

-	The choice of gene model is essential for variant downstream variant annotation: it describe genomic positions of genes and each exon-intron exact locations
-	Different gene models can give different annotations:
-	![BRCA1]({{site.url}}{{site.baseurl}}/images/brca1_var.jpg)
Variant indicated by the red dashed line can be annotated as *intronic* or *exonic* on one transcript, depending on the adopted gene model

- [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/)
- [Ensembl](https://www.ensembl.org/Homo_sapiens/Info/Index)
- [Gencode](https://www.gencodegenes.org/human/)
---
### Sequence variant nomenclature

-	Variant nomenclature should be described univocally:
- [HGVS](https://varnomen.hgvs.org/)
- [HGMC](https://www.genenames.org/)

---
### Variant class
   - [Sequence Ontology](http://www.sequenceontology.org/)
   - ![Sequence Ontology]({{site.url}}{{site.baseurl}}/images/seqOnt.png)

---
### Population sequencing db
   - [gnomAD](https://gnomad.broadinstitute.org/)
   - [ExAC](http://exac.broadinstitute.org/)
   - [1000 Genomes Project](https://www.internationalgenome.org/)
   - [NHLBI-ESP 6500 exomes](https://evs.gs.washington.edu/EVS/)
   - [dbSNP](https://www.ncbi.nlm.nih.gov/snp/)

---
### Variant-disease/gene-disease db

   - [Human Gene Mutation Database](http://www.hgmd.cf.ac.uk/ac/index.php)
   - [clinically relevant variants known in a gene](https://www.ncbi.nlm.nih.gov/clinvar/)
   - [Simple ClinVar](http://simple-clinvar.broadinstitute.org/)
   - [Online Mendelian Inheritance in Man](https://www.omim.org/)

---
### Annotation Software and tools


   - [Variant Effect Predictor](https://www.ensembl.org/info/docs/tools/vep/index.html)
   - [ANNOVAR]( http://annovar.openbioinformatics.org/en/latest/)
   - [Single Nucleotide Polymorphism Effect](http://snpeff.sourceforge.net/)
   - [KGGSeq](http://grass.cgs.hku.hk/limx/kggseq/)

- Web Interface:
   - [wAnnovar](http://wannovar.wglab.org)
