<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Data analysis and interpretation for clinical genomics</title>
        
        <link rel="stylesheet" href="/clinical_genomics/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/clinical_genomics/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/clinical_genomics/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/clinical_genomics/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/clinical_genomics/assets/css/academicons.css">
        <link rel="stylesheet" href="/clinical_genomics/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/clinical_genomics/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:title" content="Galaxy Training: Data analysis and interpretation for clinical genomics" />
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:image" content="/clinical_genomics/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        















<div class="container main-content">
    <script type="application/ld+json">
        


{}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Data analysis and interpretation for clinical genomics</h1>
        

        <div class="contributors-line">By: 

<a href="/clinical_genomics/hall-of-fame#abrusell" class="contributor-badge"><img src="https://avatars.githubusercontent.com/abrusell" alt="Alessandro Bruselles">Alessandro Bruselles</a>, <a href="/clinical_genomics/hall-of-fame#aciolfi" class="contributor-badge"><img src="https://avatars.githubusercontent.com/aciolfi" alt="Andrea Ciolfi">Andrea Ciolfi</a>, <a href="/clinical_genomics/hall-of-fame#gmauro" class="contributor-badge"><img src="https://avatars.githubusercontent.com/gmauro" alt="Gianmauro Cuccuru">Gianmauro Cuccuru</a>, <a href="/clinical_genomics/hall-of-fame#m-giuseppe" class="contributor-badge"><img src="https://avatars.githubusercontent.com/m-giuseppe" alt="Giuseppe Marangi">Giuseppe Marangi</a>, <a href="/clinical_genomics/hall-of-fame#puva" class="contributor-badge"><img src="https://avatars.githubusercontent.com/puva" alt="Paolo Uva">Paolo Uva</a>, <a href="/clinical_genomics/hall-of-fame#tommasopippucci" class="contributor-badge"><img src="https://avatars.githubusercontent.com/tommasopippucci" alt="Tommaso Pippucci">Tommaso Pippucci</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What are the specific challenges for the interpretation of sequencing data in the clinical setting?</p>
</li>
            
            <li><p>How can you annotate variants in a clinically-oriented perspective?</p>
</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Perform in-depth quality control of sequencing data at multiple levels (fastq, bam, vcf)</p>
</li>
            
            <li><p>Classify and annotate variants with information extracted from public databases for clinical interpretation</p>
</li>
            
            <li><p>Filter variants based on inheritance patterns</p>
</li>
            
            </ul>

            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 4 hours</p>
            

            

            
            

            

            <p><strong> <i class="fa fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Sep 30, 2020 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In years 2018-2019, on behalf of the Italian Society of Human Genetics (<a href="https://www.sigu.net/">SIGU</a>) an itinerant <a href="https://usegalaxy.eu/"><span class="notranslate">Galaxy</span></a>-based “hands-on-computer” training activity entitled “Data analysis and interpretation for clinical genomics” was held four times on invitation from different Italian institutions (Università Cattolica del Sacro Cuore in Rome, University of Genova, SIGU 2018 annual scientific meeting in Catania, University of Bari) and was offered to about 30 participants each time among clinical doctors, biologists, laboratory technicians and bioinformaticians. Topics covered by the course were NGS data quality check, detection of variants, copy number alterations and runs of homozygosity, annotation and filtering and clinical interpretation of sequencing results.</p>

<p>Realizing the constant need for training on NGS analysis and interpretation of sequencing data in the clinical setting, we designed an on-line <a href="https://usegalaxy.eu/"><span class="notranslate">Galaxy</span></a>-based training resource articulated in presentations and practical assignments by which students will learn how to approach NGS data quality at the level of fastq, bam and VCF files and clinically-oriented examination of variants emerging from sequencing experiments.</p>

<p>This training course is not to be intended as a tutorial on NGS pipelines and variant calling. This on-line training activity is indeed focused on data analysis for clinical interpretation. If you are looking for training on variant calling, visit this <strong><span class="notranslate">Galaxy</span></strong> tutorial on <a href="https://galaxyproject.github.io/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">Exome sequencing data analysis for diagnosing a genetic disease</a>.</p>

<blockquote>
  <blockquote class="comment">
    <h3 id="comment-sigu"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> SIGU</h3>
    <p>The <em>Italian Society of Human Genetics</em> (<strong>SIGU</strong>) was established on November 14,
1997, when the pre-existing Italian Association of Medical Genetics and the Italian
Association of Medical Cytogenetics joined.
SIGU is one of the 27 member societies of FEGS (Federation of European Genetic
Societies).
Animated by a predominant scientific spirit, SIGU wants to be reference for
all health-care issues involving human genetics in all its applications.
Its specific missions are to develop quality criteria for medical genetic
laboratories, to promote writing of guidelines in the field of human genetics
and public awareness of the role and limitations of genetic diagnostic techniques. 
SIGU coordinates activities of several working groups: Clinical Genetics,
Cytogenetics, Prenatal Diagnosis, Neurogenetics, Fingerprinting, Oncological 
Genetics, Immunogenetics, Genetic Counseling, Quality Control, Medical Genetics 
Services, Bioethics. More than 1000 medical geneticists and biologists are active
members of the society.</p>
  </blockquote>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#next-generation-sequencing" id="markdown-toc-next-generation-sequencing">Next Generation Sequencing</a>    <ol>
      <li><a href="#ngs-in-the-clinic" id="markdown-toc-ngs-in-the-clinic">NGS in the clinic</a></li>
      <li><a href="#basics-of-ngs-bioinformatic-analysis" id="markdown-toc-basics-of-ngs-bioinformatic-analysis">Basics of NGS bioinformatic analysis</a></li>
      <li><a href="#alignment" id="markdown-toc-alignment">Alignment</a></li>
      <li><a href="#post-alignment-bam-processing" id="markdown-toc-post-alignment-bam-processing">Post-alignment BAM processing</a></li>
      <li><a href="#variant-calling" id="markdown-toc-variant-calling">Variant calling</a></li>
    </ol>
  </li>
  <li><a href="#requirements" id="markdown-toc-requirements">Requirements</a></li>
  <li><a href="#datasets" id="markdown-toc-datasets">Datasets</a>    <ol>
      <li><a href="#use-cases" id="markdown-toc-use-cases">Use cases</a></li>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a>    <ol>
      <li><a href="#quality-control-of-fastq-files" id="markdown-toc-quality-control-of-fastq-files">Quality control of FASTQ files</a></li>
      <li><a href="#quality-control-of-bam-files" id="markdown-toc-quality-control-of-bam-files">Quality control of BAM files</a></li>
    </ol>
  </li>
  <li><a href="#variant-annotation" id="markdown-toc-variant-annotation">Variant annotation</a>    <ol>
      <li><a href="#gene-model" id="markdown-toc-gene-model">Gene model</a></li>
      <li><a href="#sequence-variant-nomenclature" id="markdown-toc-sequence-variant-nomenclature">Sequence variant nomenclature</a></li>
      <li><a href="#variant-class" id="markdown-toc-variant-class">Variant class</a></li>
      <li><a href="#population-sequencing-db" id="markdown-toc-population-sequencing-db">Population sequencing db</a></li>
      <li><a href="#variant-diseasegene-disease-db" id="markdown-toc-variant-diseasegene-disease-db">Variant-disease/gene-disease db</a></li>
      <li><a href="#annotation-software-and-tools" id="markdown-toc-annotation-software-and-tools">Annotation Software and tools</a></li>
      <li><a href="#annotation-and-filtering-with-snpeff" id="markdown-toc-annotation-and-filtering-with-snpeff">Annotation and filtering with SnpEFF</a></li>
    </ol>
  </li>
  <li><a href="#variant-prioritization" id="markdown-toc-variant-prioritization">Variant prioritization</a>    <ol>
      <li><a href="#variant-impact" id="markdown-toc-variant-impact">Variant impact</a></li>
      <li><a href="#variant-frequency" id="markdown-toc-variant-frequency">Variant frequency</a></li>
      <li><a href="#variant-effect-prediction-tools" id="markdown-toc-variant-effect-prediction-tools">Variant effect prediction Tools</a></li>
      <li><a href="#acmgamp-2015-guidelines" id="markdown-toc-acmgamp-2015-guidelines">ACMG/AMP 2015 guidelines</a></li>
      <li><a href="#prioritization" id="markdown-toc-prioritization">Prioritization</a></li>
      <li><a href="#gemini-for-variant-filtering" id="markdown-toc-gemini-for-variant-filtering">GEMINI for variant filtering</a></li>
      <li><a href="#candidate-variant-detection" id="markdown-toc-candidate-variant-detection">Candidate variant detection</a></li>
    </ol>
  </li>
  <li><a href="#use-cases---solutions" id="markdown-toc-use-cases---solutions">Use cases - solutions</a></li>
  <li><a href="#advanced-analysis" id="markdown-toc-advanced-analysis">Advanced analysis</a></li>
</ol>

</blockquote>

<h1 id="next-generation-sequencing">Next Generation Sequencing</h1>

<p><em>Next (or Second) Generation Sequencing</em> (NGS/SGS) is an umbrella-term covering a number of approaches to <span class="notranslate">DNA</span> sequencing that have been developed after the first, widespread and for long time most commonly used Sanger sequencing.</p>

<p><em>NGS</em> is also known as <em>Massive Parallel Sequencing</em> (MPS), a term that makes explicit the paradigm shared by all these technologies, that is to sequence in parallel a massive library of spatially separated and clonally amplified <span class="notranslate">DNA</span> templates.</p>

<p>For a comprehensive review of the different <em>NGS</em> technologies see <a href="https://www.nature.com/articles/nrg.2016.49">Goodwin et al., 2016</a>, which also includes an introduction to the third generation methods allowing sequencing of long single-molecule <span class="notranslate">reads</span>.</p>

<h2 id="ngs-in-the-clinic">NGS in the clinic</h2>

<p>In the span of less than a decade, NGS approaches have pervaded clinical laboratories revolutionizing genomic diagnostics and increasing yield and timeliness of genetic tests.</p>

<p>In the context of disorders with a recognized strong genetic contribution such as neurogenetic diseases, <em>NGS</em> has been firmly established as the strategy of choice to rapidly and efficiently diagnose diseases with a Mendelian basis. A general diagnostic <span class="notranslate">workflow</span> for these disorders currently embraces different <em>NGS</em>-based diagnostic options as illustrated in <strong>Figure 1</strong>.</p>

<hr />

<p><img src="/clinical_genomics/images/ngs_neuro_diagnostics.png" alt="ngs_neuro_diagnostics" /></p>

<p><strong>Figure 1</strong>. General <span class="notranslate">workflow</span> for genetic diagnosis of neurological diseases. (*If considering high-yield single-gene testing of more than 1–3 genes by another sequencing method, note that next-generation sequencing is often most cost-effective. †Genetic counselling is required before and after all genetic testing; other considerations include the potential for secondary findings in genomic testing, testing parents if inheritance is sporadic or recessive, and specialty referral.) From <a href="https://www.thelancet.com/journals/laneur/article/PIIS1474-4422(19)30033-X/fulltext">Rexach et al., 2019</a></p>

<hr />

<p>Currently, most common <em>NGS</em> strategies in clinical laboratories are the so-called <strong>targeted sequencing</strong> methods that, as opposed to <strong>genome sequencing</strong> covering the whole genomic sequence, focus on a pre-defined set of regions of interest (the <strong>targets</strong>). The targets can be selected by <strong>hybrid capture</strong> or <strong>amplicon sequencing</strong>, and the target-enriched libraries is then sequenced. The most popular target designs are:</p>

<ul>
  <li><strong>gene panels</strong> where the coding exons of only a clinically-relevant group of genes are targeted</li>
  <li><strong>exome sequencing</strong> where virtually all the protein-coding exons in a genome are simultaneously sequenced</li>
</ul>

<h2 id="basics-of-ngs-bioinformatic-analysis">Basics of NGS bioinformatic analysis</h2>

<p>Apart from the different width of the target space in exome and gene panels, these two approaches usually share the same experimental procedure for NGS library preparation. After clonal amplification, the fragmented and adapter-ligated <span class="notranslate">DNA</span> templates are sequenced from both ends (<strong>paired-end</strong> sequencing) of the <em>insert</em> to produce short <span class="notranslate">reads</span> in opposite (<strong>forward</strong> and <strong>reverse</strong>) orientation.</p>

<p>Bioinformatic analysis of NGS data usually follows a general three-step <span class="notranslate">workflow</span> to variant detection. Each of these three steps is marked by its “milestone” file type containing sequence data in different formats and metadata describing sequence-related information collected during the analysis step that leads to generation of that file.</p>

<table>
  <thead>
    <tr>
      <th>NGS <span class="notranslate">workflow</span> step</th>
      <th>File content</th>
      <th>File format</th>
      <th>File Size (individual exome)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sample to <span class="notranslate">reads</span></td>
      <td>Unaligned <span class="notranslate">reads</span> and qualities</td>
      <td><strong>fastQ</strong></td>
      <td>gigabytes</td>
    </tr>
    <tr>
      <td><span class="notranslate">Reads</span> to alignments</td>
      <td>Aligned <span class="notranslate">reads</span> and metadata</td>
      <td><strong>BAM</strong></td>
      <td>gigabytes</td>
    </tr>
    <tr>
      <td>Alignments to variants</td>
      <td>Genotyped variants and metadata</td>
      <td><strong>VCF</strong></td>
      <td>megabytes</td>
    </tr>
  </tbody>
</table>

<p>Here are the different formats explained:</p>

<ul>
  <li><strong><a href="https://en.wikipedia.org/wiki/FASTQ_format">fastQ</a></strong> (sequence with quality): the <em>de facto</em> standard for storing the output of high-throughput sequencing machines
    <ul>
      <li>Usually not inspected during data analysis</li>
    </ul>
  </li>
  <li><strong><a href="https://samtools.github.io/hts-specs/SAMv1.pdf">BAM</a></strong> (binary sequence alignment/map): the most widely used TAB-delimited file format to store alignments onto a reference sequence
    <ul>
      <li>Aligned <span class="notranslate">reads</span></li>
    </ul>
  </li>
  <li><strong><a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf">VCF</a></strong> (variant call format): the standard TAB-delimited format for genotype information associated with each reported genomic position where a variant call has been recorded</li>
</ul>

<p>Another useful file format is <a href="https://www.ensembl.org/info/website/upload/bed.html"><span class="notranslate">BED</span></a>, to list genomic regions of interest such as the exome or panel targets.</p>

<p>The steps of the <strong><em><span class="notranslate">reads</span>-to-variants</em></strong> <span class="notranslate">workflow</span> can be connected through a bioinformatic <strong>pipeline</strong> (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5429012/">Leipzig et al., 2017</a>), consisting in read alignments, post-alignment BAM processing and variant calling.</p>

<h2 id="alignment">Alignment</h2>

<p>As generated by the sequencing machines, <em>paired-end</em> <span class="notranslate">reads</span> are written to two <em>fastQ</em> files in which <em>forward</em> and <em>reverse</em> <span class="notranslate">reads</span> are stored separately together with their qualities. <em>FastQ</em> files are taken as input files by tools (the <strong>aligners</strong>) that align the <span class="notranslate">reads</span> onto a reference genome. One of the most used aligners is <a href="http://bio-bwa.sourceforge.net/">BWA</a> among the many that have been developed (<strong>Figure 2</strong>).</p>

<hr />

<p><img src="/clinical_genomics/images/aligners_timeline.png" alt="aligners_timeline" /></p>

<p><strong>Figure 2</strong>. Aligners timeline 2001-2012 (from <a href="https://academic.oup.com/bioinformatics/article/28/24/3169/245777">Fonseca et al., 2012</a>)</p>

<hr />

<p>During the bioinformatic process, <em>paired-end</em> <span class="notranslate">reads</span> from the two separate <em>fastQ</em> files are re-connected in the alignment, where it is expected that they will:</p>
<ul>
  <li>map to their correct location in the genome</li>
  <li>be as distant as the insert size of the fragment they come from</li>
  <li>be in opposite orientations
a combination which we refer to as <strong>proper pairing</strong>. All these data about <em>paired-end</em> <span class="notranslate">reads</span> <span class="notranslate">mapping</span> are stored in the BAM file and can be used to various purposes, from alignment quality assessment to structural variant detection.</li>
</ul>

<p>In <strong>Figure 3</strong>, the <a href="http://software.broadinstitute.org/software/igv/">Integrative Genomic Viewer (IGV)</a> screenshot of an exome alignment data over two adjacent <em>ASXL1</em> exons is shown. Pink and violet bars are <em>forward</em> and <em>reverse</em> <span class="notranslate">reads</span>, respectively. The thin grey link between them indicates that they are <em>paired-end</em> <span class="notranslate">reads</span>. The stack of <span class="notranslate">reads</span> is concentrated where exons are as expected in an exome, and the number of read bases covering a given genomic location <em>e</em> (depicted as a hill-shaped profile at the top of the figure) defines the <strong>depth of coverage (DoC)</strong> over that location:</p>

<p><em>DoC</em><sub>e</sub>=<em>number of read bases over e/genomic length of e</em></p>

<hr />

<p><img src="/clinical_genomics/images/coverage.png" alt="coverage" />
<strong>Figure 3</strong>. Exome data visualization by <a href="http://software.broadinstitute.org/software/igv/"><em>IGV</em></a></p>

<hr />

<h2 id="post-alignment-bam-processing">Post-alignment BAM processing</h2>

<p>Regarding post-alignment <em>pipelines</em>, the most famous for germline SNP and InDel calling is probably that developed as part of the GATK toolkit (<strong>Figure 4</strong>).</p>

<hr />

<p><img src="/clinical_genomics/images/gatk_germline_snps_indels.png" alt="gatk_germline_pipeline" /></p>

<p><strong>Figure 4</strong>. After-alignment pipeline for germline SNP and InDel variant calling according to <a href="https://software.broadinstitute.org/gatk/best-practices/workflow?id=11145">GATK Best Practices</a></p>

<hr />

<p>According to GATK best practices, in order to be ready for variant calling the BAM file should undergo the following processing:</p>
<ul>
  <li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.4.0/picard_sam_markduplicates_MarkDuplicates.php">marking duplicate <span class="notranslate">reads</span></a> to flag (or discard) <span class="notranslate">reads</span> that are mere optical or PCR-mediated duplications of the actual <span class="notranslate">reads</span></li>
  <li><a href="https://software.broadinstitute.org/gatk/documentation/article?id=11081">recalibrating base quality scores</a> to correct known biases of the native base-related qualities
While GATK BAM processing is beyond doubt important to improve data quality, it has to be noticed that it is not needed to obtain variant calls and that non GATK-based pipelines may not use it or may use different quality reparametrization schemes. Duplicate flagging or removal is not recommended in <em>amplicon</em> sequencing experiments.</li>
</ul>

<h2 id="variant-calling">Variant calling</h2>

<p>The process of variant detection and genotyping is performed by <em>variant callers</em>. These tools use probabilistic approaches to collect evidence that non-reference read bases accumulating over a given locus support the presence of a variant, usually differing in algorithms, filtering strategies, recommendations (<a href="https://doi.org/10.1038/srep43169">Sandmann et al., 2017</a>). To be confident that a variant is a true event, its supporting evidence should be significantly stronger than chance; e.g. the C&gt;T on the left of the screenshot in <strong>Figure 5</strong> is supported by all its position-overlapping <span class="notranslate">reads</span>, claiming for a variant. In contrast, the C&gt;A change on the right of the screenshot is seen only once over many <span class="notranslate">reads</span>, challenging its interpretation as a real variant. In fact, <span class="notranslate">DNA</span> variants that occur in germ cells (i.e., <strong>germline/constitutional variants</strong> that can be passed on to offspring) are either diploid/biallelic, so expected alte<span class="notranslate">rna</span>tive allele frequency is 50% for a heterozygous change. On the other hand, if only a smaller subset of aligned <span class="notranslate">reads</span> indicates variation, that could result from technology bias or be a <strong>mosaicism</strong>, i.e. an individual which harbour two or more populations of genetically distinct cells as a result of postzygotic mutation. Postzygotic <em>de novo</em> mutations may result in <strong>somatic mosaicism</strong> (potentially causing a less severe and/or variable phenotype compared with the equivalent constitutive mutation) <strong>and/or germline mosaicism</strong> (hence enabling transmission of a pathogenic variant from an unaffected parent to his affected offspring) (<a href="https://doi.org/10.1038/nrg3424">Biesecker et al., 2013</a>). To identify mosaicism, a probabilistic approach should consider deviation of the proband variant allele fraction (VAF, defined as the number of alte<span class="notranslate">rna</span>tive <span class="notranslate">reads</span> divided by the total read depth) from a binomial distribution centred around 0.5.</p>

<hr />

<p><img src="/clinical_genomics/images/igv_variant.png" alt="igv_screenshot_variant" /></p>

<p><strong>Figure 5</strong>. Variant visualization by <em>IGV</em></p>

<hr />

<p>The GATK variant calling pipeline first produces a <strong>genomic VCF (<a href="https://gatkforums.broadinstitute.org/gatk/discussion/4017/what-is-a-gvcf-and-how-is-it-different-from-a-regular-vcf">gVCF</a>)</strong>, whose main difference with <em>VCF</em> is that it records all sites in a target whether there is a variant or not, while <em>VCF</em> contains only information for variant sites, preparing multiple samples for <strong>joint genotyping</strong> and creation of a <strong>multi-sample</strong> <em>VCF</em> whose variants can undergo quality <strong>filtering</strong> in order to obtain the final set of quality-curated variants ready to be annotated.</p>

<p>In <span class="notranslate">downstream</span> analyses, annotations can be added to <em>VCF</em> files themselves or information in <em>VCF</em> files can be either annotated in TAB- or comma- deimited files to be visually inspected for <em>clinical</em> variant searching or used as input to <strong>prioritization</strong> programs.</p>

<h1 id="requirements">Requirements</h1>

<p>This tutorial is based on the <a href="https://galaxyproject.org/"><span class="notranslate">Galaxy</span></a> platform,
therefore a basic knowledge of <span class="notranslate">Galaxy</span> is required to get most out of the course.
In particular, we’ll use the European <span class="notranslate">Galaxy</span> server running at <a href="https://usegalaxy.eu">https://use<span class="notranslate">galaxy</span>.eu</a>.</p>

<p>Registration is <strong>free</strong>, and you get access to <strong>250GB</strong> of disk space for your analysis.</p>

<ol>
  <li>Open your browser. We recommend Chrome or Firefox (please don’t use Internet Explorer or Safari).</li>
  <li>Go to <a href="https://usegalaxy.eu">https://use<span class="notranslate">galaxy</span>.eu</a>
    <ul>
      <li>If you have previously registered on this server just log in:
        <ul>
          <li>On the top menu select: User -&gt; Login</li>
          <li>Enter your user/password</li>
          <li>Click Submit</li>
        </ul>
      </li>
      <li>If you haven’t registered on this server, you’ll need to do now.
        <ul>
          <li>On the top menu select: User -&gt; Register</li>
          <li>Enter your email, choose a password, repeat it and add a one word name (lower case)</li>
          <li>Click Submit</li>
        </ul>
      </li>
    </ul>
  </li>
</ol>

<p>To familiarize with the <span class="notranslate">Galaxy</span> interface (e.g. working with histories, importing dataset),
we suggest to follow the <a href="https://galaxyproject.github.io/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html"><span class="notranslate">Galaxy</span> 101</a> tutorial.</p>

<h1 id="datasets">Datasets</h1>

<p>Input datasets used in this course are available:</p>
<ul>
  <li>at <a href="https://zenodo.org/record/3531578">Zenodo</a>, an open-access repository developed under the European OpenAIRE program and operated by CERN</li>
  <li>as <em>Shared Data Libraries</em> in <a href="https://usegalaxy.eu/library/list"><span class="notranslate">Galaxy</span></a>: <em><a href="https://usegalaxy.eu/library/list#folders/F3d08bb711e4e3b26"><span class="notranslate">Galaxy</span> courses / Sigu</a></em></li>
</ul>

<h2 id="use-cases">Use cases</h2>

<p>We selected some case studies for this tutorial.
We suggest you to start with a simple case (e.g. XXX) for the first run of the tutorial,
and repeat it with the more complex ones.
At the end of the page you will be able to compare your candidate variants
with the list of true pathogenic variants.</p>

<table>
  <thead>
    <tr>
      <th>Case</th>
      <th>Number of individuals</th>
      <th>Description FIXME</th>
      <th>…</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Trio 1</td>
      <td>3</td>
      <td>…</td>
      <td>…</td>
    </tr>
    <tr>
      <td>Trio 2</td>
      <td>3</td>
      <td>…</td>
      <td>…</td>
    </tr>
    <tr>
      <td>Trio 3</td>
      <td>3</td>
      <td>…</td>
      <td>…</td>
    </tr>
    <tr>
      <td>Trio 4</td>
      <td>3</td>
      <td>…</td>
      <td>…</td>
    </tr>
  </tbody>
</table>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name (e.g. Clinical genomics)</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fa fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fa fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import files from <a href="https://zenodo.org/record/3531578">Zenodo</a> or Shared Data Library:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3531578/files/CNV_case.bam
https://zenodo.org/record/3531578/files/CNV_control.bam
https://zenodo.org/record/3531578/files/CNV_TruSeq_Chr2.<span class="notranslate">bed</span>
https://zenodo.org/record/3531578/files/HighQuality_<span class="notranslate">Reads</span>.fastq.gz
https://zenodo.org/record/3531578/files/LowQuality_<span class="notranslate">Reads</span>.fastq.gz
https://zenodo.org/record/3531578/files/Panel_alignment.bam
https://zenodo.org/record/3531578/files/Panel_target_regions.<span class="notranslate">bed</span>
https://zenodo.org/record/3531578/files/Sample1.all_exons.hg19.vcf
</code></pre></div>      </div>
      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fa fa fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <p>The same files may be available on the <span class="notranslate">Galaxy</span> server
through a <em>Shared Data Libraries</em> in 
<em><a href="https://usegalaxy.eu/library/list#folders/F3d08bb711e4e3b26"><span class="notranslate">Galaxy</span> courses/Sigu</a></em>.
You may prefer to import the data directly from there.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-note"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note</h3>
        <p>All the files are based on <code class="language-plaintext highlighter-rouge">hg19</code> reference genome which is
available with pre-built indexes for widely used tools such as 
<em>bwa-mem</em> <strong>and</strong> <em>samtools</em> by selecting <code class="language-plaintext highlighter-rouge">hg19</code> version as an option under
<em>“(Using) reference genome”</em>).</p>
      </blockquote>
    </li>
    <li>
      <p>In case you import datasets from Zenodo, check that all datasets in your history
have their datatypes assigned correctly, and fix it when necessary.
For example, to assign <span class="notranslate">BED</span> datatype do the following:</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fa fa fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span></code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Rename the datasets</p>

      <p>For datasets uploaded via a link, <span class="notranslate">Galaxy</span> will use the link
as the dataset name. In this case you may rename datasets.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="quality-control">Quality control</h1>

<p>In-depth quality control (QC) of data generated during an NGS experiment is crucial for an accurate interpretation of the results. For example an accurate QC could help in identifying poor quality experiments, sequence contamination or genomic regions with low sequence coverage, and all these factors have a large impact on the <span class="notranslate">downstream</span> processing.</p>

<p>Most of the programs used during an NSG <span class="notranslate">workflow</span> do not include steps for quality control, therefore artifacts needs to be detected using ad-hod developed tools for QC. The table  summarizes the main tools available at <a href="https://usegalaxy.eu">https://use<span class="notranslate">galaxy</span>.eu</a> for quality checking, at each step of the analysis.</p>

<table>
  <thead>
    <tr>
      <th>NGS <span class="notranslate">workflow</span> step</th>
      <th>File format</th>
      <th>Tools for quality control</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Sample to <span class="notranslate">reads</span></td>
      <td>fastQ</td>
      <td><strong>FastQC</strong></td>
    </tr>
    <tr>
      <td><span class="notranslate">Reads</span> to alignments</td>
      <td>BAM</td>
      <td>General statistics: <strong>bam.io.bio</strong>, <strong>samtools</strong>; target coverage: <strong>Picard CollectHSMetrics</strong>; per base coverage depth: <strong><span class="notranslate">bed</span>tools</strong></td>
    </tr>
    <tr>
      <td>Alignments to variants</td>
      <td>VCF</td>
      <td><strong>vcf.io.bio</strong></td>
    </tr>
  </tbody>
</table>

<hr />

<h2 id="quality-control-of-fastq-files">Quality control of FASTQ files</h2>

<p>Before starting the analysis <span class="notranslate">workflow</span>, you should identify possible issues that could affect alignment and variant calling. This first step of quality control is based on the raw sequence data (fastQ) generated by the sequencer. Common issues with sequence quality can be easily addressed by further processing your original sequences to trim or remove low-quality <span class="notranslate">reads</span>. In presence of severe artefacts you should consider to repeat the experiment instead of starting the <span class="notranslate">downstream</span> analysis that will generate poor quality results, according to the rule ‘garbage in, garbage out’.</p>

<p>Here we’ll use the <strong>FastQC</strong> software for a standard quality check, using the two FASTQ files <em>HighQuality_<span class="notranslate">Reads</span>.fastq</em> and <em>LowQuality_<span class="notranslate">Reads</span>.fastq</em>.</p>

<p><em>FastQC</em> is relatively easy to use. The output of <strong>FastQC</strong> consists of multiple modules analysing a specific aspect of the quality of the data. A detailed help can be found in the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">help manual</a>.</p>

<p>The names of the modules are preceded by an icon that reflects the quality of the data, and indicates whether the results of the module are:</p>
<ul>
  <li>normal (green)</li>
  <li>slightly abnormal (orange)</li>
  <li>very unusual (red)</li>
</ul>

<blockquote class="comment">
  <h3 id="comment-note-on-fastqc-interpretation"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note on FastQC interpretation</h3>
  <p>These evaluations must be taken in the context of what you are expecting from your
dataset. For <strong>FastQC</strong> a <em>normal</em> sample includes random sequences with high diversity. 
If your experiment generates biased libraries (e.g. low complexity
libraries) you should interpret the report with attention. In general, you should
concentrate on the icons different from green and try to understand the reasons for
this behaviour.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computing-sequence-quality-with-fastqc"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computing sequence quality with FastQC</h3>
  <ol>
    <li>
      <p>Run <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on your fastq datasets <em>HighQuality_<span class="notranslate">Reads</span>.fastq</em> and
<em>LowQuality_<span class="notranslate">Reads</span>.fastq</em>. You can select both datasets with the  <strong>Multiple datasets</strong>
option.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>

      <p>For each input file you will get two datasets, one with the raw QC statistics and
another with an HTML report with figures.</p>
    </li>
    <li>Using the <strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> software, you can aggregate multiple raw
<strong>FastQC</strong> output in one unique report. This helps in comparing multiple samples at
the same time in order to quickly identify low quality samples that will be displayed
as outliers.
      <ul>
        <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code>
          <ul>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: the two <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect <strong>MultiQC</strong> report</li>
  </ol>

  <p>For a detailed explanation of the different analysis modules of <strong>FastQC</strong> you may refer
to the <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality control</a> tutorial.</p>

  <blockquote class="question">
    <h3 id="question-questions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>Based on the <strong>MultiQC</strong> report, check which modules highlight differences in sequence quality betwen the two datasets</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="quality-control-of-bam-files">Quality control of BAM files</h2>

<h3 id="fast-quality-check-with-bamiobioio">Fast quality check with bam.iobio.io</h3>

<p>BAM files are binary files containing information on the sequences aligned onto a reference genome. Exploring BAM files you can address several questions, e.g.:</p>

<ul>
  <li>which is the amount of duplicated sequences? For non PCR-free protocols, it should be &lt; 15%. Duplicated sequences are not used in <span class="notranslate">downstream</span> analysis to identify variants, therefore should be kept at a minimum to avoid waste of reagents.</li>
  <li>which is the fraction of unmapped <span class="notranslate">reads</span>? It should be &lt; 2%. If higher, you should ask why so many <span class="notranslate">reads</span> are not properly mapped onto the reference genome. One possible reason could be sample contamination.</li>
</ul>

<p>In <span class="notranslate">Galaxy</span>, BAM files can be explored using the <em>bam.iobio.io</em> web app. Leveraging on random subsampling of <span class="notranslate">reads</span>, <strong>bam.iobio.io</strong> quickly draws several quality control reports (<strong>Figure 6</strong>).</p>

<p>On top of each plot, clicking on the question mark you can open a window with a detailed explanation of the expected output. The number of <span class="notranslate">reads</span> sampled is shown at the top-right of the page, and can be increased by clicking on the arrow.</p>

<p><img src="/clinical_genomics/images/bamiobio.png" alt="bam_iobio" /></p>

<p><strong>Figure 6</strong>. BAM quality control using <em>bam.iobio.io</em></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computing-bam-quality-with-bamiobioio"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computing BAM quality with bam.iobio.io</h3>
  <ol>
    <li>
      <p>Run <strong>bam.iobio.io</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on a BAM dataset. To start <strong>bam.iobio.io</strong>
click on the link <code class="language-plaintext highlighter-rouge">diplay at bam.iobio.io</code> in the dataset section. Please note
that the link will be visible only for datasets with the appropriate database 
field set to <code class="language-plaintext highlighter-rouge">hg19</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
        <ul>
          <li>Click on the <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Database/Build</strong> field</li>
          <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">hg19</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>Which is the amount of duplicated sequences?</li>
          <li>And the fraction of aligned <span class="notranslate">reads</span>?</li>
        </ol>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h3 id="coverage-metrics-with-picard">Coverage metrics with Picard</h3>

<p><strong>Collect Hybrid Selection (HS) Metrics</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool computes a set of metrics that are specific for sequence datasets generated through hybrid-selection, a commonly used protocol to capture specific sequences for targeted experiments such as panels and exome sequencing.</p>

<p>In order to run this tool you need a file with the aligned sequences in BAM format, and files with the intervals corresponding to bait and target regions. These files can be generally obtained from the website of the kit manufacturer.</p>

<blockquote class="comment">
  <h3 id="comment-note-1"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note</h3>
  <p>Please note that interval files are generally available as <span class="notranslate">BED</span> files,
and must be converted in Picard interval_list format using Picard’s
<strong><span class="notranslate">Bed</span>ToInterval</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> before running CollectHsMetrics - see
the hands on below for details.</p>
</blockquote>

<p>Metrics generated by CollectHsMetrics are grouped into three classes:</p>
<ul>
  <li>Basic sequencing metrics: genome size, the number of <span class="notranslate">reads</span>, the number of aligned <span class="notranslate">reads</span>, the number of unique <span class="notranslate">reads</span>, etc.</li>
  <li>Metrics for evaluating the performance of the wet-lab protocol: number of bases <span class="notranslate">mapping</span> on/off/near baits, number of bases <span class="notranslate">mapping</span> on target, etc.</li>
  <li>Metrics for coverage estimation: mean bait coverage, mean and median target coverage, the percentage of bases covered at different coverage (e.g. 1X, 2X, 10X, 20X, …), the percentage of filtered bases, etc.</li>
</ul>

<p>For a detailed description of the output see <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#HsMetrics">Picard’s CollectHsMetrics</a></p>

<p>In the next tutorial we will compute hybrid-selection metrics for BAM files containing 
aligned sequences from an exome sequencing experiment.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computing-bam-statistics-with-picard-collecthsmetrics"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computing BAM statistics with Picard CollectHsMetrics</h3>

  <ol>
    <li>
      <p>Before computing the statistics, we first need to convert the <span class="notranslate">BED</span> files
with bait and target regions, in Picard interval_list format.
Remember that you can select multiple datasets
with the <strong>Multiple datasets</strong> option.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>

      <p>Run <strong><span class="notranslate">Bed</span>ToIntervalList</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to convert <span class="notranslate">BED</span> files.</p>
      <ul>
        <li><em>“Load picard dictionary file from?”</em>: <code class="language-plaintext highlighter-rouge">Local cache</code>
          <ul>
            <li>In <em>“Use dictionary from the list”</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg19</code></li>
          </ul>
        </li>
        <li><em>“Select coordinate dataset or dataset collection?”</em>: your <span class="notranslate">BED</span> file to be converted</li>
      </ul>
    </li>
    <li>
      <p>Run <strong>CollectHsMetrics</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> using as input the BAM files and the intervals
in Picard interval_list format, corresponding to the bait and target regions,
generated in the previous step.</p>
    </li>
    <li>Use <strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to aggregate CollectHsMetrics output in one unique
report to facilitate the comparison across multiple samples.
      <ul>
        <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">Picard</code>
          <ul>
            <li>In <em>“Picard output”</em>
              <ul>
                <li><em>“Type of Picard output?”</em>: <code class="language-plaintext highlighter-rouge">HS Metrics</code></li>
                <li><i class="fa fa-files-o" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Picard output”</em>: the output of <strong><em>CollectHsMetrics</em></strong>
<i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect <strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> report</p>

      <blockquote class="question">
        <h3 id="question-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>Which is the average target coverage?</li>
          <li>And the fraction of bases covered at least 10X?</li>
        </ol>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="variant-annotation">Variant annotation</h1>
<p>After the generation of a high-quality set of mapped read pairs, we can proceed to call different classes of <span class="notranslate">DNA</span> variants.
Users interested in germline variant calling can refer to related <span class="notranslate">Galaxy</span>’s tutorials, e.g. <a href="https://galaxyproject.github.io/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">Exome sequencing data analysis for diagnosing a genetic disease</a>.
Once called, variants (SNPs and InDels) need to be annotated.</p>

<p>We want to know for example if a variant is located in a gene, if it’s in the coding portion of that gene, if it causes an aminoacid substitution, if that substitution is deleterious for the encoded protein function.</p>

<p><strong>Variant annotation</strong> is the process of attaching technical and/or biological information from multiple available source (e.g. <strong>public databases</strong>) to variants</p>

<hr />
<h2 id="gene-model">Gene model</h2>

<p>The choice of gene model is essential for variant <span class="notranslate">downstream</span> variant annotation: it describe genomic positions of genes and each exon-intron exact locations</p>

<p>Different gene models can give different annotations:
<img src="/clinical_genomics/images/brca1_var.jpg" alt="BRCA1" /></p>

<p><strong>Figure 1.</strong> Variant indicated by the red dashed line can be annotated as <em>intronic</em> or <em>exonic</em> (on one of the UCSC transcript variants), depending on the adopted gene model:</p>

<table>
  <thead>
    <tr>
      <th>Source</th>
      <th>Description FIXME</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a></td>
      <td> </td>
    </tr>
    <tr>
      <td><a href="https://www.ensembl.org/Homo_sapiens/Info/Index">Ensembl</a></td>
      <td> </td>
    </tr>
    <tr>
      <td><a href="https://www.gencodegenes.org/human/">Gencode</a></td>
      <td> </td>
    </tr>
  </tbody>
</table>

<hr />
<h2 id="sequence-variant-nomenclature">Sequence variant nomenclature</h2>

<p>Variant nomenclature should be described univocally:</p>

<table>
  <thead>
    <tr>
      <th>Source</th>
      <th>Description FIXME</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><a href="https://varnomen.hgvs.org/">HGVS</a></td>
      <td> </td>
    </tr>
    <tr>
      <td><a href="https://www.genenames.org/">HGMC</a></td>
      <td> </td>
    </tr>
  </tbody>
</table>

<h2 id="variant-class">Variant class</h2>

<p>Sequence features used in biological sequence annotation should be defined 
 using the <a href="http://www.sequenceontology.org/">Sequence Ontology</a>
 <img src="/clinical_genomics/images/seqOnt.png" alt="Sequence Ontology" /></p>

<h2 id="population-sequencing-db">Population sequencing db</h2>

<ul>
  <li><a href="https://gnomad.broadinstitute.org/">gnomAD</a></li>
  <li><a href="http://exac.broadinstitute.org/">ExAC</a></li>
  <li><a href="https://www.internationalgenome.org/">1000 Genomes Project</a></li>
  <li><a href="https://evs.gs.washington.edu/EVS/">NHLBI-ESP 6500 exomes</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a></li>
</ul>

<h2 id="variant-diseasegene-disease-db">Variant-disease/gene-disease db</h2>

<ul>
  <li><a href="http://www.hgmd.cf.ac.uk/ac/index.php">Human Gene Mutation Database</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/clinvar/">clinically relevant variants known in a gene</a></li>
  <li><a href="http://simple-clinvar.broadinstitute.org/">Simple ClinVar</a></li>
  <li><a href="https://www.omim.org/">Online Mendelian Inheritance in Man</a></li>
  <li><a href="https://sites.google.com/site/jpopgen/dbNSFP"><strong>dbNSFP</strong></a>, a big database of curated annotations and precomputed functional predictions for all potential non-synonymous and splice-site single-nucleotide variants in the human genome</li>
</ul>

<h2 id="annotation-software-and-tools">Annotation Software and tools</h2>
<p><strong>Local installation</strong>:</p>
<ul>
  <li><a href="https://www.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor</a></li>
  <li><a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a></li>
  <li><a href="http://snpeff.sourceforge.net/">Single Nucleotide Polymorphism Effect</a></li>
  <li><a href="http://grass.cgs.hku.hk/limx/kggseq/">KGGSeq</a></li>
</ul>

<p><strong>Web interface</strong>:</p>
<ul>
  <li><a href="http://wannovar.wglab.org">wAnnovar</a></li>
  <li><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP">VEP</a></li>
</ul>

<hr />

<h2 id="annotation-and-filtering-with-snpeff">Annotation and filtering with SnpEFF</h2>

<p>For variant annotations we’ll use SnpEff, a software for genomic variant annotation and functional effect prediction.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-annotations-with-snpeff"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant annotations with SnpEff</h3>
  <ol>
    <li><strong>SnpEff eff</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (not the one for the SARS-CoV-2 pipeline)
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sequence changes (SNPs, MNPs, InDels)”</em>: the
uploaded <strong>VCF</strong> file</li>
        <li><em>“Input format”</em>: <code class="language-plaintext highlighter-rouge">VCF</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
        <li><em>“Genome source”</em>: <code class="language-plaintext highlighter-rouge">Locally installed snpEff database</code>
          <ul>
            <li><em>“Genome”</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> (or a similarly named option)</li>
          </ul>
        </li>
        <li><em>“Produce Summary Stats”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Two output file will be created:</p>
<ol class="hands_on">
  <li>a <em>Summary Stats</em> HTML report, with general metrics
such as the distribution of variants across gene features;</li>
  <li>a VCF file with annotations of variant effects added to the INFO
column.</li>
</ol>

<h1 id="variant-prioritization">Variant prioritization</h1>

<p>Once annotated, variants need to be filtered and prioritized. The number of variants
returned by genomic sequencing varies from tens of thousands (WES) to millions (WGS).</p>

<p><strong>No universal filters are available, they depend on the experimental features</strong></p>

<h2 id="variant-impact">Variant impact</h2>

<p>First of all you usually want to filter variants by consequence on the encoded protein, keeping those which have an higher impact on protein:</p>

<ul>
  <li>Missense</li>
  <li>Nonsense</li>
  <li>Splice sites</li>
  <li>Frameshift indels</li>
  <li>Inframe indels</li>
</ul>

<h2 id="variant-frequency">Variant frequency</h2>

<ul>
  <li>Common variants are unlikely associated with a clinical condition</li>
  <li>A rare variant will probably have a higher functional effect on the protein</li>
  <li>Frequency cut-off have to be customized on each different case</li>
  <li>Typical cut-offs: 1% - 0.1%</li>
  <li>Allele frequencies may differ a lot between different populations</li>
</ul>

<h2 id="variant-effect-prediction-tools">Variant effect prediction Tools</h2>

<ul>
  <li>Tools that predict consequences of amino acid substitutions on protein function</li>
  <li>
    <p>They give a score and/or a prediction in terms of “Tolerated”, “Deleterious” (SIFT) or “Probably Damaging”, “Possibly Damaging”, “Benign” (Polyphen2)</p>

    <ul>
      <li>fitCons</li>
      <li>GERP++</li>
      <li>SIFT</li>
      <li>PolyPhen2</li>
      <li>CADD</li>
      <li>DANN</li>
      <li>Condel</li>
      <li>fathmm</li>
      <li>MutationTaster</li>
      <li>MutationAssessor</li>
      <li>REVEL</li>
    </ul>
  </li>
</ul>

<h2 id="acmgamp-2015-guidelines">ACMG/AMP 2015 guidelines</h2>

<p>The <em>American College of Medical Genetics</em> and the <em>Association for Molecular Pathology</em> published guidelines for the interpretation of sequence variants in May of 2015 <a href="https://www.nature.com/articles/gim201530">(Richards S. et al, 2015)</a>. This report describes updated standards and guidelines for classifying sequence variants by using criteria informed by expert opinion and experience</p>

<ul>
  <li>28 evaluation criteria for the clinical interpretation of variants. Criteria falls into 3 sets:
    <ul>
      <li>pathogenic/likely pathogenic (P/LP)</li>
      <li>benign/likely benign (B/LB)</li>
      <li>variant of unknown significance (VUS)</li>
    </ul>
  </li>
  <li><a href="http://wintervar.wglab.org"><strong>Intervar</strong></a>: software for automatical interpretation of the 28 criteria. Two major steps:</li>
  <li>automatical interpretation by 28 criteria</li>
  <li>manual adjustment to re-interpret the clinical significance</li>
</ul>

<h2 id="prioritization">Prioritization</h2>

<p>Phenotype-based prioritization tools are methods working by comparing the phenotypes of a patient with gene-phenotype known associations.</p>

<ul>
  <li><a href="http://phenolyzer.wglab.org/">Phenolyzer</a>: Phenotype Based Gene Analyzer, a tool to prioritize genes based on user-specific disease/phenotype terms</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-prioritization"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant prioritization</h3>

  <p>Using knowledge gained on <a href="http://0.0.0.0:4000/clinical_genomics/lectures/04.db.html">Genomic databases and variant annotation</a> section, try to <strong><em>annotate</em></strong>, <strong><em>filter</em></strong> and <strong><em>prioritize</em></strong> an example exome variant data, using two disease terms</p>

  <ul>
    <li>
      <p>Download file Sample1.all_exons.hg19.vcf from the <em>dataset</em>, as learned in the <a href="http://0.0.0.0:4000/clinical_genomics/lectures/01.home.html">Home</a> section</p>
    </li>
    <li>
      <p>Go to <a href="http://wannovar.wglab.org/">wANNOVAR</a></p>
    </li>
    <li>
      <p>Use the vcf file as <strong>input file</strong> and <em>hearing loss</em> and <em>deafness autosomal recessive</em>, as disease terms to prioritize results</p>
    </li>
    <li>
      <p>Choose <em>rare recessive Mendelian disease</em> as Disease Model in the <strong><em>Parameter Settings</em></strong> section</p>
    </li>
    <li>
      <p>Provide an istitutional email address and submit the Job</p>
    </li>
    <li>
      <p>… <em>wait for results</em> …</p>

      <p>In the results page you can navigate and download results.
 Click <strong><em>Show</em></strong> in the <em>Network Visualization</em> section to see
 Phenolyzer prioritization results</p>
    </li>
  </ul>
</blockquote>

<h2 id="gemini-for-variant-filtering">GEMINI for variant filtering</h2>

<p>Now, we’ll use the VCF file annotated with SnpEff to filter variants considering
the relationship between family members.
For this purpose we’ll use GEMINI, a framework including different modules for analysis of human variants.
First, we need to inform GEMINI about the relationship between the samples and their phenotypes (affected vs not affected).
This information is stored in a <strong>pedigree file</strong> in PED format.
In next Hands-on you’ll learn how to manually generate a pedigree file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-gemini-pedigree-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a GEMINI pedigree file</h3>
  <ol>
    <li>
      <p>Create an example PED-formatted pedigree file for a trio:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#family_id    name        pate<span class="notranslate">rna</span>l_id    mate<span class="notranslate">rna</span>l_id    sex    phenotype
Fam_A         father_ID   0              0              1      1
Fam_A         mother_ID   0              0              2      1
Fam_A         proband_ID  father_ID      mother_ID      1      2
</code></pre></div>      </div>

      <p>and set its datatype to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-file"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new file</h3>

        <ul>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the file contents into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code></p>
          </li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </blockquote>

      <blockquote class="warning">
        <h3 id="warning-remember-those-sample-names"><i class="fa fa-warning" aria-hidden="true"></i><span class="visually-hidden">warning</span> Remember those sample names</h3>

        <p>Names in the pedigree file should match the sample names in 
your VCF file in order to be recongnized by GEMINI. If names are different,
samples will not be recognized and therefore you will not be able to filter
variants by patterns of genetic inheritance.</p>

      </blockquote>

      <blockquote class="details">
        <h3 id="details-more-on-ped-files"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More on PED files</h3>

        <p>The PED format is explained in the help section of <strong>GEMINI load</strong>
<i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531972-PED-Pedigree-format">here</a></p>

        <p>Take a moment and try to understand the information that is encoded in
the PED dataset we are using here.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>Next, in order to formulate queries to extract variants matching your selection criteria, 
variants and their annotations need to be stored in a format accepted by GEMINI.
This task is accomplished by the <strong>GEMINI load</strong> tool, which accepts as input your SnpEFF
SnpEff annotated VCF file together with the pedigree file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-gemini-database"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a GEMINI database</h3>

  <ol>
    <li><strong>GEMINI load</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="fa fa-file-o" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“VCF dataset to be loaded in the GEMINI database”</em>:
the output of <strong>SnpEff eff</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“The variants in this input are”</em>: <code class="language-plaintext highlighter-rouge">annotated with snpEff</code></li>
        <li>
          <p><em>“This input comes with genotype calls for its samples”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>Our examples VCf include genotype calls.</p>
        </li>
        <li><em>“Choose a gemini annotation source”</em>: select the latest available annotations snapshot (most likely, there will be only one)</li>
        <li><em>“Sample and family information in PED format”</em>: the pedigree file
prepared above</li>
        <li><em>“Load the following optional content into the database”</em>
          <ul>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“GERP scores”</em></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“CADD scores”</em></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Gene tables”</em></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Sample genotypes”</em></li>
            <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“variant INFO field”</em></li>
          </ul>

          <p><strong>Checked</strong> the following:</p>
          <ul>
            <li>
              <p><em>“only variants that passed all filters”</em></p>

              <p>This retains only high quality variants, e.g variants with the value in the 
FILTER column equals to PASS</p>
            </li>
          </ul>

          <p>Leave <strong>unchecked</strong> the following:</p>
          <ul>
            <li>
              <p><em>“Genotype likelihoods (sample PLs)”</em></p>

              <p>Our VCFs does not contain these values</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This generates a GEMINI-specific dataset, which can only
be processed with other GEMINI tools. In fact, every analysis with a GEMINI tool
starts with the GEMINI database obtained by <strong>GEMINI load</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<blockquote class="details">
  <h3 id="details-the-gemini-suite-of-tools"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> The GEMINI suite of tools</h3>

  <p>The GEMINI framework is composed by a large number of utilities.</p>

  <p>The <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
demonstrates the use of the <strong>GEMINI annotate</strong> and <strong>GEMINI query</strong> tools,
and tries to introduce some essential bits of GEMINI’s SQL-like syntax.</p>

  <p>For a thorough explanation of all GEMINI tools and functionality visit 
the <a href="https://gemini.readthedocs.io">GEMINI documentation</a>.</p>
</blockquote>

<h2 id="candidate-variant-detection">Candidate variant detection</h2>

<p>Here you’ll learn how to use <strong>GEMINI inheritance pattern</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to report all variants
fitting any specific inheritance model. You’ll be able to select any of the following inheritance patterns:</p>

<ul>
  <li>Autosomal recessive</li>
  <li>Autosomal dominant</li>
  <li>X-linked recessive</li>
  <li>X-linked dominant</li>
  <li>Autosomal de-novo</li>
  <li>X-linked de-novo</li>
  <li>Compound heterozygous</li>
  <li>Loss of heterozygosity (LOH) events</li>
</ul>

<p>Below is how you can perform the query for inherited autosomal recessive
variants. Feel free to run analogous queries for other types of variants that
you think could plausibly be causative for your case study.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-variants-by-inheritance-pattern"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering variants by inheritance pattern</h3>
  <ol>
    <li><strong>GEMINI inheritance pattern</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><em>“GEMINI database”</em>: the GEMINI database of annotated variants; output
of <strong>GEMINI load</strong> <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Your assumption about the inheritance pattern of the phenotype of interest”</em>:
e.g. <code class="language-plaintext highlighter-rouge">Autosomal recessive</code>
          <ul>
            <li><i class="fa fa-plus-square-o" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Additional constraints on variants”</em>
              <ul>
                <li>
                  <p><em>“Additional constraints expressed in SQL syntax”</em>:
<code class="language-plaintext highlighter-rouge">impact_severity != 'LOW'</code></p>

                  <p>This will remove variants with <em>low impact severity</em> (<em>i.e.</em>, silent
mutations and variants outside coding regions).
Leave this box empty to report all variants independently of their impact.</p>
                </li>
              </ul>
            </li>
            <li>
              <p><em>“Include hits with less convincing inheritance patterns”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>Account for errors in phenotype assignment - meaningful for large families</p>
            </li>
            <li>
              <p><em>“Report candidates shared by unaffected samples”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>Account for incomplete penetrance - meaningful for large families</p>
            </li>
          </ul>
        </li>
        <li>
          <p><em>“Family-wise criteria for variant selection”</em>: keep default settings</p>

          <p>This section is not useful when you have data from just one family.</p>
        </li>
        <li>In <em>“Output - included information”</em>
          <ul>
            <li><em>“Set of columns to include in the variant report table”</em>:
<code class="language-plaintext highlighter-rouge">Custom (report user-specified columns)</code>
              <ul>
                <li><em>“Choose columns to include in the report”</em>:
                  <ul>
                    <li><i class="fa fa-check-square-o" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“alte<span class="notranslate">rna</span>tive allele frequency (max_aaf_all)”</em></li>
                  </ul>
                </li>
                <li><em>“Additional columns (comma-separated)”</em>:
<code class="language-plaintext highlighter-rouge">chrom, start, ref, alt, impact, gene, clinvar_sig,
clinvar_disease_name, clinvar_gene_phenotype, rs_ids</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="details">
        <h3 id="details-clinvar-annotations"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> ClinVar annotations</h3>

        <p><strong>clinvar_sig</strong> and <strong>clinvar_disease_name</strong> annotations refer to the particular variant,
<strong>clinvar_gene_phenotype</strong> provides information about the gene harbouring the variant.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>From the output of <strong>GEMINI inheritance pattern</strong>, can you identify the most likely
candidate variant?</p>
</blockquote>

<blockquote class="details">
  <h3 id="details-more-gemini-usage-examples"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More GEMINI usage examples</h3>

  <p>While only demonstrating command line use of GEMINI, the following tutorial
slides may give you additional ideas for variant queries and filters:</p>

  <ul>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Intro-To-Gemini.pdf">Introduction to GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-DeNovo-Tutorial.pdf">Identifying <em>de novo</em> mutations with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Recessive-Tutorial.pdf">Identifying recessive gene candidates with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Dominant-Tutorial.pdf">Identifying dominant gene candidates with GEMINI</a></li>
  </ul>
</blockquote>

<blockquote class="comment">
  <h3 id="trophy-congratulations"><i class="fa fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> <strong>Congratulations!</strong></h3>
  <p>You successfully completed the tutorial.
Below you will find the true patogenic variants for the cases 
presented in this tutorial.</p>
</blockquote>

<h1 id="use-cases---solutions">Use cases - solutions</h1>
<p>FIXME</p>

<h1 id="advanced-analysis">Advanced analysis</h1>
<p>If you are interested in CNV analysis, identification of RoH…FIXME you can move to the 
<a href="https://sigu-training.github.io/clinical_genomics/advanced.html">Advanced tutorial</a></p>

<h1 class="no_toc" id="contributors">Contributors</h1>

<ul>
  <li><a href="https://www.aosp.bo.it/content/curriculum?E=154659">Tommaso Pippucci</a> - Sant’Orsola-Malpighi University Hospital, Bologna, Italy</li>
  <li><a href="https://www">Alessandro Bruselles</a> - Istituto Superiore di Sanità, Rome, Italy</li>
  <li><a href="http://www.ospedalebambinogesu.it">Andrea Ciolfi</a> - Ospedale Pediatrico Bambino Gesù, IRCCS, Rome, Italy</li>
  <li><a href="https://gmauro.github.io">Gianmauro Cuccuru</a> - Albert Ludwigs University, Freiburg, Germany</li>
  <li><a href="http://www">Giuseppe Marangi</a> - Institute of Genomic Medicine, Fondazione Policlinico Universitario A. Gemelli IRCCS, Università Cattolica del Sacro Cuore, Roma, Italy</li>
  <li><a href="https://www.researchgate.net/profile/Paolo_Uva">Paolo Uva</a> - IRCCS G. Gaslini, Genoa, Italy</li>
</ul>



                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Focus on clinical interpretation of variants.</p>
</li>
                            
                            <li><p>Provides real uses cases.</p>
</li>
                            
                            <li><p>Use public tools and free annotation servers.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Alessandro Bruselles, Andrea Ciolfi, Gianmauro Cuccuru, Giuseppe Marangi, Paolo Uva, Tommaso Pippucci, 2020 <b>Data analysis and interpretation for clinical genomics (Galaxy Training Materials)</b>. <a href="/clinical_genomics/">/clinical_genomics/</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{-,
    author = "Alessandro Bruselles and Andrea Ciolfi and Gianmauro Cuccuru and Giuseppe Marangi and Paolo Uva and Tommaso Pippucci",
    title = "Data analysis and interpretation for clinical genomics (Galaxy Training Materials)",
    year = "2020",
    month = "09",
    day = "30"
    url = "\url{/clinical_genomics/}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us feedback on this content!</h3>
            <p>To give us feedback about these materials, or to get in touch with us, post a message in
               the forum <a href="https://groups.google.com/d/forum/sigu-training">https://groups.google.com/d/forum/sigu-training</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            The layout of this page is based on the <a href="https://training.galaxyproject.org/">Galaxy Training Resources</a> collection.
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/sigu-training/clinical_genomics/blob/master/index.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/clinical_genomics/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/clinical_genomics/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/clinical_genomics/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/clinical_genomics/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/clinical_genomics/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/clinical_genomics/assets/js/main.js"></script>

    <script type="text/javascript" src="/clinical_genomics/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    
</html>
